### PR TITLE
Adds clarification & examples to demystify Transforms.

### PR DESCRIPTION
**Pull request by DougReeder**
**Description**
This expands the (somewhat terse) documentation

### CODE REVIEW

File: docs/api/transforms.md [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# Transforms
+# Transforms API
```
Comment (dylans) [inline docs/api/transforms.md:1]: I don't think we call the other API sections as API, so I would either change all of them or revert this.

File: docs/concepts/04-transforms.md [hunk 2]
```diff
@@ -4,6 +4,40 @@ Slate's data structure is immutable, so you can't modify or delete nodes directl
...
+  match: node =>
+    !Editor.isEditor(node) &&
+    node.children?.every(child => Editor.isBlock(editor, child)),
+  mode: 'all', // also the Editor's children
+})
+```
+
+Non-standard operations (or debugging/tracing which Nodes will be affected by a set of NodeOptions) may require using
+`Editor.nodes` to create a JavaScript Iterator of NodeEntries and a for..of loop to act.
+Here's code to replace all image elements with their alt text:
```
Comment (dylans) [inline docs/concepts/04-transforms.md:22]:
```suggestion
For example, to replace all image elements with their alt text:
```

**Review by dylans — CHANGES_REQUESTED** [review]
A couple of small suggestions, and then I think this is good to land. Thanks @DougReeder !

File: docs/api/transforms.md (hunk 1 above)
Comment (DougReeder) [reply docs/api/transforms.md:1]: I've added "API" to all of the API documents, to distinguish them from the Concepts documents.

**Review by dylans — APPROVED** [review]

**Pull request merged**